### ISSUE TITLE
`new` operator does not work well with full qualifier

### ISSUE THREAD

**Issue by felixhao28** [atom/language-java#181]
### Description

```java
public class A {
    void f() {
        java.io.File myFile = java.io.File.create(folderPath);
        java.io.File myFile = new java.io.File(folderPath);
    }
}
```

![image](https://user-images.githubusercontent.com/2417391/49370572-86afec80-f72f-11e8-8322-35371ea7f637.png)

**Expected behavior:** [What you expect to happen]

"java.io" after `new` is not recognized as part of class name.

**Comment by sadikovi** [comment]
Can you include the language-java version that you are testing against?

**Comment by felixhao28** [comment]
d48e7137f6634cc094f092b68e6b110f51f47630

**Comment by felixhao28** [comment]
might be related, `extends` does not work well with full qualifiers either:

```java
class RenderType extends com.jfinal.plugin.ehcache.RenderType {
    public static final int BEETL_RENDER = 6;
}
```
![image](https://user-images.githubusercontent.com/2417391/49423438-7b61cd00-f7d2-11e8-9149-3a07a725100b.png)

**Comment by sadikovi** [comment]
This looks like a regression, because it was fixed at some point, if I remember correctly. I will investigate.

**Comment by Vigilans** [comment]
The regression is caused in these lines:

https://github.com/atom/language-java/blob/master/grammars/java.cson#L1059-L1062

the constraint `must be followed by a variable name` prevents `java.io.File` recognized as a type.

**Comment by sadikovi** [comment]
That's not correct. It does get recognised, the regex skips the first item in fully qualified class name - just need to fix that.

**Comment by Vigilans** [comment]
It would be fixed naturally if `java.io.File` is recoginized as a type and fall into `object-type` scope to be parsed...

@sadikovi If you agree types in constructor call should be rendered as type, you can take a look at [this branch](https://github.com/Vigilans/language-java/compare/vigilans/array-initializer...Vigilans:vigilans/new-type) at your convenience, which is come up to solve #216.

It is simple enough by just doing following things:
* Add `(?=\\()` match to `object-type`, to allow type to be recognized in a constructor call.
* Remove `function-call` and `parens` from `anonymous-classes-and-new`, and patch a `(...)`  matching for constructor calls.
* Fix the udpated scopes in test by:
  * Replace all 'meta.function-call.java', 'entity.name.function.java' with 'storage.type.java';
  * Remove 'meta.function-call.java' before 'punctuation.definition.parameters.*.bracket.round.java'
